examples/defaulter-gen/generators/defaulter.go

```
@@ -401,6 +436,75 @@ func newCallTreeForType(existingDefaulters, newDefaulters defaulterFuncMap) *cal
+	var prev *types.Type
+	for prev != t {
+		prev = t
+		if t.Kind == types.Alias {
+			t = t.Underlying
+		}
+	}
+	return t
+}
+
+func mustEnforceDefault(t *types.Type, omitEmpty bool) (interface{}, error) {
+	switch t.Kind {
+	case types.Pointer, types.Map, types.Slice, types.Array, types.Interface:
+		return nil, nil
+	case types.Struct:
+		return map[string]interface{}{}, nil
+	case types.Builtin:
+		if !omitEmpty {
+			if zero, ok := typeZeroValue[t.String()]; ok {
+				return zero, nil
+			} else {
+				return nil, fmt.Errorf("please add type %v to typeZeroValue struct", t)
+			}
+
+		}
+		return nil, nil
+	default:
+		return nil, fmt.Errorf("not sure how to enforce default for %v", t.Kind)
+	}
+}
+
+func populateDefaultValue(node *callNode, t *types.Type, tags string, commentLines []string) *callNode {
+	t = resolveAliasType(t)
+	defaultMap := extractDefaultTag(commentLines)
+	if len(defaultMap) == 1 {
+		var defaultValue interface{}
+		if err := json.Unmarshal([]byte(defaultMap[0]), &defaultValue); err != nil {
+			klog.Fatalf("Failed to unmarshal default: %v", err)
+		}
+
+		omitEmpty := strings.Contains(reflect.StructTag(tags).Get("json"), "omitempty")
+		if enforced, err := mustEnforceDefault(t, omitEmpty); err != nil {
+			klog.Fatal(err)
+		} else if enforced != nil {
+			if defaultValue != nil {
+				if reflect.DeepEqual(defaultValue, enforced) {
+					// If the default value annotation matches the default value for the type,
+					// do not generate any defaulting function
+					return node
+				} else {
+					enforcedJSON, _ := json.Marshal(enforced)
+					klog.Fatalf("Invalid default value (%#v) for non-pointer/non-omitempty. If specified, must be: %v", defaultValue, string(enforcedJSON))
+				}
+			}
+		}
+
+		// callNodes are not automatically generated for primitive types. Generate one if the callNode does not exist
+		if node == nil {
+			node = &callNode{}
+		}
+
+		node.defaultValue = defaultMap[0]
+	} else if len(defaultMap) > 1 {
```

Can you return that error earlier, that'll unnest everything by one-level.